File: app/lib/model/transcription.dart
import 'package:app/model/detection.dart';
import 'package:uuid/uuid.dart';

class Transcription {
  late String _id;
  late Detection _detection;
  late String _value;
  late DateTime _transcripedAt;

  Transcription(this._detection, this._value, this._transcripedAt,
      {String? id}) {
    _id = id ?? const Uuid().v4();
  }

  String get id => _id;
  Detection get detection => _detection;
  String get value => _value;
  DateTime get transcripedAt => _transcripedAt;
}

File: web_app/api/model/transcription.py
import time
import datetime
import uuid
from .detection import Detection


class Transcription:
    uid:str
    detection:list[Detection]
    value:str
    transcripedAt:time

    def __init__(self, detection:Detection, value:str, uid:str=None, transcripedAt:time=None):
        if uid == None:
            self.uid = str(uuid.uuid4())
        else:
            self.uid = uid
        if time == None:
            self.time = time.time()
        else:
            self.time = time
        self.detection = detection
        self.value = value
        self.transcripedAt = transcripedAt

    def toDict(self):
        detectionList = []
        for det in self.detections:
            detectionList.append(det.toDict())
        return {
            "uid":self.uid,
            "detection":detectionList,
            "value":str(self.value),
            "transcripedAt": str(datetime.datetime.fromtimestamp(self.transcripedAt))
        }